src/backend/executor/nodeDML.c

```diff
@@ -116,6 +115,13 @@ ExecDML(DMLState *node)
 		 * es_result_relations will contain the only relation.
 		 */
 		node->ps.state->es_result_relation_info = relInfo;
+
+		/*
+		 * Perform partition selection for in place update
+		 */
+		if (isUpdate && !AttributeNumberIsValid(plannode->actionColIdx))
```

Comment:
We don't need to perform partition selection in case of updating leaf-partition. Also take ADBDEV-4207 into account in terms of avoiding extra ri_PartCheckMap allocation